spec/models/group_spec.rb

```diff
@@ -386,4 +386,22 @@
     end
   end
 
+  describe '#average_completion_score' do
+    let(:team) { create(:group) }
+    let(:subteam) { create(:group, parent: team) }
+
+    it 'returns 0 when no one exists in subtree' do
+      expect(team.average_completion_score).to be 0
+    end
+
+    it 'returns average completion score of all people in group and subtree' do
+      create(:membership, group: team, person: create(:person))
+      create(:membership, group: team, person: create(:person))
+      create(:membership, group: subteam, person: create(:person, city: 'Wherever'))
+      expect(team.average_completion_score).to be_between(47, 49)
```

Comment:
A factory user will have a specific score (44 i think) but additional attributes will bump this up, and the score is an average for the team - deliberatley kept it rough as score criteria could change and this test just ensures it is testing the average for the subtree